rwth_upper_body_skeleton_random_walk/src/upper_body_skeleton_random_walk_node.cpp

```diff
@@ -0,0 +1,338 @@
+#include <ros/ros.h>
+#include <image_transport/image_transport.h>
+#include <cv_bridge/cv_bridge.h>
+#include <sensor_msgs/image_encodings.h>
+#include <std_msgs/Int8.h>
+#include <fstream>
+#include <string>
+#include <iostream>
+#include <message_filters/subscriber.h>
+#include <image_transport/subscriber_filter.h>
+#include <message_filters/time_synchronizer.h>
+#include "upper_body_detector/UpperBodyDetector.h"
+#include <message_filters/synchronizer.h>
+#include <message_filters/sync_policies/approximate_time.h>
+#include <QImage>
+#include <QPainter>
+#include <vector>
+#include "forest_utilities.h"
+#include "bodypose_regressor_utilities.h"
+#include <opencv/cv.h>
+#include <opencv/highgui.h>
+#include "rwth_upper_body_skeleton_random_walk/SkeletonDetector.h"
+
+unsigned counter = 0;
+unsigned detection = 0;
+unsigned im_counter = 0;
+using namespace rwth_upper_body_skeleton_random_walk;
+
+image_transport::Publisher pub_result_image;
+ros::Publisher pub_3d_positions;
+ros::ServiceClient upper_body_skeleton_client;
+
+int joint_connections[9] = {-1,0,1,1,1,3,4,5,6};
+unsigned bone_colors[8][3] = {{0,0,255},{0,0,255},{255,0,255},{255,0,0},{0,255,0},{255,255,0},{255,0,255},{255,0,0}};
+
+FOREST f(9,640,480);
+
+//struct body_pose_inputs
+//{
+//    float depth_image[307200];
+//    float depths[50000];
+//    unsigned pixels_x[50000];
+//    unsigned pixels_y[50000];
+  //  unsigned start_x, end_x;
+  //  unsigned end_x, end_y; 
+//};
+
+
+//for debugging only
+void frames_to_file(const cv::Mat &im,std::vector<float> &bounding_box)
+{
+    ROS_INFO("Writing frame : %d", (int) im_counter);
+    //counter++;
+    size_t N;
+    char filepath[200];
+    std::ofstream myfile;
+
+    //cv::Mat out;
+    //cv::cvtColor(im, out, CV_RGB2BGR);
+    im_counter++;
+    sprintf(filepath,"/home/mmp/catkin_ws/data/images/image_%d.txt",im_counter);
```

Comment:
I assume this is never called in the live system as it says for debugging only?! Otherwise, no hard coded paths please.